src/polymer/expression-scanner.ts

```diff
@@ -120,6 +126,17 @@ export abstract class DatabindingExpression {
       return;
     }
     let expression = expressionStatement.expression;
+    if (limitation === 'identifierOnly') {
+      if (expression.type !== 'Identifier') {
+        this.warnings.push(this._validationWarning(
+            `Expected just a name here, not an expression`, expression));
+      }
+    } else if (limitation === 'callExpression') {
+      if (expression.type !== 'CallExpression') {
+        this.warnings.push(this._validationWarning(
```

Comment:
Moved to a method and a switch/case implementation that ensures all limitation values are checked or it's a compile time error.